mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,429 @@
+# Author: Mainak Jas <mainak.jas@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+import numpy as np
+import warnings
+
+from ...utils import logger, verbose, check_version
+from ..utils import _read_segments_file
+from ..meas_info import _empty_info, create_info
+from ..base import _BaseRaw, _check_update_montage
+from ...channels.montage import Montage
+from ...epochs import _BaseEpochs
+from ...event import read_events
+from ...externals.six import string_types
+
+
+def _check_fname(fname):
+    """Check if the file extension is valid.
+    """
+    fmt = str(op.splitext(fname)[-1])
+    if fmt == '.dat':
+        raise NotImplementedError(
+            'Old data format .dat detected. Please update your EEGLAB '
+            'version and resave the data in .fdt format')
+    elif fmt != '.fdt':
+        raise IOError('Expected .fdt file format. Found %s format' % fmt)
+
+
+def _check_mat_struct(fname):
+    """Check if the mat struct contains 'EEG'.
+    """
+    if not check_version('scipy', '0.12'):
+        raise RuntimeError('scipy >= 0.12 must be installed for reading EEGLAB'
+                           ' files.')
+    from scipy import io
+    mat = io.whosmat(fname, struct_as_record=False,
+                     squeeze_me=True)
+    if 'ALLEEG' in mat[0]:
+        msg = ('Loading an ALLEEG array is not supported. Please contact'
+               'mne-python developers for more information.')
+        raise NotImplementedError(msg)
+    elif 'EEG' not in mat[0]:
+        msg = ('Unknown array in the .set file.')
+        raise ValueError(msg)
+
+
+def _rescale_data(info, data):
+    """Put data to EEG scale.
+    """
+    for idx, ch in enumerate(info['chs']):
+        if len(data.shape) == 2:
+            data[idx] *= ch['cal']
+        elif len(data.shape) == 3:
+            data[:, idx] *= ch['cal']
+    return data
+
+
+def _to_loc(ll):
+    """Check if location exists.
+    """
+    if isinstance(ll, (int, float)) or len(ll) > 0:
+        return ll
+    else:
+        return 0.
+
+
+def _get_info(eeg, montage):
+    """Get measurement info.
+    """
+    info = _empty_info(sfreq=eeg.srate)
+    info['nchan'] = eeg.nbchan
+
+    # add the ch_names and info['chs'][idx]['loc']
+    path = None
+    if len(eeg.chanlocs) > 0:
+        ch_names, pos = [], []
```

Comment:
I see a few of this, I think we started to initialize standard library classes using the functions, not the syntax, e.g. list, dict, to make our code more accessible for python newcomers.